main.rb

```diff
@@ -1,22 +1,68 @@
-# Create class Person
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
   def initialize(age, id, name: 'Unknown', parent_permission: true)
-    @id = id
+    super()
     @name = name
     @age = age
+    @id = id
     @parent_permission = parent_permission
   end
 
   def can_use_services?
     @age >= 18 || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age?
     @age < 18
   end
 end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    original_name = @nameable.correct_name
+    original_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    original_name = @nameable.correct_name
+    if original_name.length > 10
+      original_name[0, 10]
+    else
+      original_name
+    end
+  end
+end
```

Comment:
- [OPTIONAL] Great job decorating your classes. I would love to suggest that, it would be great if you could modularize your classes into separate files. You did a great work. :handshake: